complaints/models.py

```diff
@@ -116,25 +118,30 @@ def validate_timings(self, avail_start_time, avail_end_time):
         if self.bhavan in self.GIRLS_BHAVANS:
             for start_time, end_time in self.GIRLS_TIME_LIMITS:
                 if avail_start_time >= start_time and avail_end_time <= end_time:
-                    break
+                    return
             else:
                 raise ValidationError(
                     "Girls' Hostel is only open for limited time slots. Your given time does not fit in them"
                 )
+        for start_time, end_time in self.TECHNICIAN_WORK_HOURS:
+            if avail_start_time >= start_time and avail_end_time <= end_time:
+                break
         else:
-            if avail_end_time < time(
-                avail_start_time.hour + 1,
-                avail_start_time.minute,
-                avail_start_time.second,
-            ):
-                raise ValidationError("Available time is less than one hour.")
+            raise ValidationError("Availble time not in technicians' working hours")
+        if self.avail_date == date.today():
+            if avail_start_time < datetime.now().time():
+                raise ValidationError("Available time before complaint registration.")
+        if avail_end_time < time(
+            avail_start_time.hour + 1, avail_start_time.minute, avail_start_time.second
+        ):
+            raise ValidationError("Available time is less than one hour.")
```

Comment:
In every validation error add the message  " Complaint not registered .....  Try again! "